src/local/butler/lint.py

```diff
@@ -15,10 +15,140 @@
+  extension = os.path.splitext(file_path)[1]
+  if (filename not in _LICENSE_CHECK_FILENAMES and
+      extension not in _LICENSE_CHECK_EXTENSIONS):
+    return
+
+  path_directories = file_path.split(os.sep)
+  for directory in _LICENSE_CHECK_IGNORE_DIRECTORIES:
+    if directory in path_directories:
+      return
+
+  source_filename = os.path.basename(file_path)
+  for check_filename in _LICENSE_CHECK_IGNORE_FILENAMES:
+    if check_filename == source_filename:
+      return
+
+  if _LICENSE_CHECK_STRING in open(file_path).read():
+    return
+
+  print 'Missing license header for %s.' % file_path
+  sys.exit(1)
+
+
+def py_import_order(file_path):
+  """Validate that python imports are alphabetized."""
+
+  def _validate_block(import_block):
+    """Ensure that a single block is ordered properly."""
+    if not import_block:
+      return []
+
+    sorted_import_block = sorted(import_block, key=lambda i: i.lower())
+    if sorted_import_block == import_block:
+      return []
+
+    return ['\n'.join(sorted_import_block)]
+
+  def _validate_imports(data):
+    """Test that a file's contents are ordered properly."""
+    imports = []
+    from_imports = []
+    corrected_import_blocks = []
+    for line in data.splitlines():
+      if line.startswith('import '):
+        imports.append(line)
+      else:
+        corrected_import_blocks += _validate_block(imports)
+        imports = []
+
+      if line.startswith('from '):
+        from_imports.append(line)
+      else:
+        corrected_import_blocks += _validate_block(from_imports)
+        from_imports = []
+
+    # Though rare, if a file ends with an import we must still validate them.
+    corrected_import_blocks += _validate_block(imports)
+    corrected_import_blocks += _validate_block(from_imports)
+
+    if not corrected_import_blocks:
+      return
+
+    suggestions = '\n\n--------\n\n'.join(corrected_import_blocks)
+    print('File {filename} has non-alphabetized import blocks. '
+          'Suggested order:\n\n{suggestions}').format(
+              filename=file_path, suggestions=suggestions)
+    sys.exit(1)
+
+  with open(file_path) as handle:
+    _validate_imports(handle.read())
+
+
+def py_test_init_check(file_path):
+  """Check test directory has a __init__.py file. Otherwise, the test does not
+  execute at all."""
+  if not file_path.endswith(_PY_TEST_SUFFIX):
+    return
+
+  test_directory = os.path.dirname(file_path)
+  if _PY_INIT_FILENAME not in os.listdir(test_directory):
+    print 'Missing {filename} file in test directory {dir}.'.format(
+        filename=_PY_INIT_FILENAME, dir=test_directory)
+    sys.exit(1)
+
+
+def yaml_validate(file_path):
+  """Run yaml validation for the given source path."""
+  if os.path.basename(file_path) in _YAML_EXCEPTIONS:
+    return
+
+  try:
+    with open(file_path) as handle:
+      yaml.safe_load(handle.read())
+  except Exception as e:
+    print 'Failed yaml validation for %s.\n\n%s' % (file_path, e)
+    sys.exit(1)
```

Comment:
Why stop linting if one yaml file is invalid? See my comment on lint.py:177

Reply:
see comment above.